### PR TITLE
Ingestion chunking [partial fix for #1434]

### PR DESCRIPTION
**Pull request by abhidg**
This PR adds chunking support for CSV files to solve the backfill issue. Other file types (currently JSON and XLSX) are not chunked.

* Change return type of `retrieve_content()`. `retrieve_content()` now returns list of `(filename, s3 object key)`
    tuples instead of a single tuple corresponding to the chunks. Only for CSVs, `retrieve_content_csv()` is called which performs chunking, with each chunk limited to `CSV_CHUNK_BYTES`. No chunking is performed for JSON and XLSX.
 * `invoke_parser()` is called in a loop with the s3 object keys.
 
What's the best practice to test the output of retrieval_content_csv()? Currently there are no content tests (such as checking filetype). As I see it, one would need to have a fake local URL to pass to requests. Any pointers welcome!

Also, what happens if one particular chunk fails at some stage, say uploading to S3. Do we get missing data for that day? How would we know of this failure?

Currently CSV_CHUNK_BYTES is set to 100 MB. We need to test whether this is an appropriate value. @AnyaLindstromBattle mentioned S3 processing being variable speed, so we would need to adjust this.

### CODE REVIEW

**Comment by abhidg** [comment]
This PR only works for CSV files which do not have newlines in content as it uses newlines for record boundaries. I will have a look at Pandas chunking, which might be a better idea for general CSV files.

**Review by iamleeg — CHANGES_REQUESTED** [review]
My review comments are one nit and one question, so not worth blocking this over, and this PR has introduced a test failure in CI so that's worth addressing before merge. Thanks!

**Comment by iamleeg** [comment]
Still a test failing on CI as a result of this change.

File: ingestion/functions/retrieval/retrieval.py [hunk 1]
```diff
@@ -102,18 +104,91 @@ def raw_content(url: str, content: bytes) -> io.BytesIO:
...
+
+    This method chunks the CSV file to avoid timeouts in the ingestion functions.
+    Chunking is controlled by the `chunk_bytes` parameter, which defaults to 10 MiB.
+    """
+    csv_header = None  # Assume no header by default
+    try:
+        print(f"Downloading CSV content from {url}")
+        headers = {"user-agent": "GHDSI/1.0 (http://ghdsi.org)"}
+        r = requests.get(url, headers=headers)
+        r.raise_for_status()
```
[the remark this answers is not in the document]
Comment (abhidg) [reply ingestion/functions/retrieval/retrieval.py:120]: Good idea, I've kept the same method as in `retrieve_content` for consistency, I can take a look after the merge

**Review by calremmel — APPROVED** [review]
This looks great, a big step forward. Thanks @abhidg

**Pull request merged**